spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc

```diff
@@ -0,0 +1,223 @@
+
+[[arch-analytics]]
+== Analytics
+Spring Cloud Data Flow is aware of certain Sink applications that will write counter data to Redis and provides an REST endpoint to read counter data.  The types of counters supported are
+
+* link:https://github.com/spring-cloud/spring-cloud-stream-app-starters/tree/master/metrics/spring-cloud-starter-stream-sink-counter[Counter] - Counts the number of messages it receives, optionally storing counts in a separate store such as redis.
+* link:https://github.com/spring-cloud/spring-cloud-stream-app-starters/tree/master/metrics/spring-cloud-starter-stream-sink-field-value-counter[Field Value Counter] - Counts occurrences of unique values for a named field in a message payload
+* link:https://github.com/spring-cloud/spring-cloud-stream-app-starters/tree/master/metrics/spring-cloud-starter-stream-sink-aggregate-counter[Aggregate Counter] - Stores total counts but also retains the total count values for each minute, hour day and month.
+
+It is important to note that the timestamp that is used in the aggregate counter can come from a field in the message itself so that out of order messages are properly accounted.
+
+ifdef::omit-tasks-docs[]
+[[arch-task]]
+== Task Applications
+
+The Spring Cloud Task programming model provides 
+
+* Persistence of the Task’s lifecycle events and exit code status.
+* Lifecycle hooks to execute code before or after a task execution.
+* Emit task events to a stream (as a source) during the task lifecycle.
+* Integration with Spring Batch Jobs.
+endif::omit-tasks-docs[]
+
+[[arch-data-flow-server]]
+== Data Flow Server
+
+[[arch-data-flow-server-endpoints]]
+=== Endpoints
+
+The Data Flow Server uses an embedded servlet container and exposes REST endpoints for creating, deploying, undeploying, and destroying streams and tasks, querying runtime state, analytics, and the like. The Data Flow Server is implemented using Spring’s MVC framework and the link:https://github.com/SpringSource/spring-hateoas[Spring HATEOAS] library to create REST representations that follow the HATEOAS principle.
+
+.The Spring Cloud Data Flow Server 
+image::{dataflow-asciidoc}/images/data-flow-server-arch.png[The Spring Cloud Data Flow Server Architecture, scaledwidth="100%"]
```

Comment:
Thanks for catching the typo.  In the current scheme, it is a catch-22, as you can't render the image until the commit is done to master. The reason is we are including this into a bunch ofher service impl docs.